node/src/components/linear_chain_sync.rs

```diff
@@ -392,6 +389,14 @@ impl<I: Clone + PartialEq + 'static> LinearChainSync<I> {
             }
         }
     }
+
+    fn current_block(&self) -> Option<&BlockHeader> {
+        match &self.state {
+            State::SyncingTrustedHash { current_block, .. } => Option::as_ref(&*current_block),
+            State::SyncingDescendants { current_block, .. } => Some(&*current_block),
```

Comment:
How come you can do `Option::as_ref(…)` above and `Some(…)` here? Can you use the same version in both places?

Reply:
I think the problem I ran into is that `Box` has `as_ref`, too, so if you call it on a `Box<Option<_>>`, it will call the `Box` one and return an `&Option<_>`. `SyncingDescendants`' `current_block` is not an `Option` anymore.